From e74617600e472a58c47007ebab27131d063ecacf Mon Sep 17 00:00:00 2001
From: Henrique de Moraes Holschuh <henrique@nic.br>
Date: Mon, 2 Sep 2019 14:22:27 -0300
Subject: [PATCH 5/5] ujail: add O_CLOEXEC flag to open() call

if we forget to close() in the future, this prevent fd leak.

(backport note: uclibc in Chaos Calmer lacks O_CLOEXEC for no
 good reason)

Signed-off-by: Etienne CHAMPETIER <champetier.etienne@gmail.com>
Backported-by: Henrique de Moraes Holschuh <henrique@nic.br>
(backported from commit 7646217c0d1775bc95cceb14f5c975ac7d157f05)
---
 jail/elf.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/jail/elf.c b/jail/elf.c
index cbb3051..48bc9b3 100644
--- a/jail/elf.c
+++ b/jail/elf.c
@@ -88,6 +88,9 @@ static int elf_open(char **dir, char *file)
 	if (fd == -1)
 		fd = open(file, O_RDONLY);
 
+	if (fd != -1)
+		(void) fcntl(fd, F_SETFD, FD_CLOEXEC);
+
 	return fd;
 }
 
-- 
2.17.1

